docs/topics/jvm/jvm-spring-boot-using-crudrepository.md

```diff
@@ -0,0 +1,97 @@
+</microformat>
+
+In this part, you will migrate the service layer to use the [Spring Data](https://docs.spring.io/spring-data/commons/docs/current/api/org/springframework/data/repository/CrudRepository.html) `CrudRepository` instead of `JdbcTemplate` for database access.
+_CrudRepository_ is a Spring Data interface for generic [CRUD](https://en.wikipedia.org/wiki/Create,_read,_update_and_delete) operations on a repository of a specific type.
+It provides several methods out of the box for interacting with a database.
+
+## Update your application
+
+First, you need to adjust the `Message` class for work with the `CrudRepository` API:
+
+1. Add the `@Table` annotation to the `Message` class to declare mapping to a database table.  
+   Add the `@Id` annotation before the `id` field.
+
+    > These annotations also require additional imports.
+    >  
+    {type="note"}
+
+    ```kotlin
+    import org.springframework.data.annotation.Id
+    import org.springframework.data.relational.core.mapping.Table
+    
+    @Table("MESSAGES")
+    data class Message(@Id var id: String?, val text: String)
+    ```
+    
+    Besides adding the annotations, you also need to make the `id` mutable (`var`) for the reasons of how `CrudRepository` works when inserting the new objects to the database.
+
+2. Declare an interface for the `CrudRepository` that will work with the `Message` data class:
+
+    ```kotlin
+    import org.springframework.data.repository.CrudRepository
+    
+    interface MessageRepository : CrudRepository<Message, String>
+    ```
+
+3. Update the `MessageService` class. It will now call to the `MessageRepository` instead of executing SQL queries:
+
+    ```kotlin
+    @Service
+    class MessageService(val db: MessageRepository) {
+        fun findMessages(): List<Message> = db.findAll().toList()
+    
+        fun findMessageById(id: String): List<Message> = db.findById(id).toList()
+    
+        fun save(message: Message) {
+            db.save(message)
+        }
+    
+        fun <T : Any> Optional<out T>.toList(): List<T> =
+            if (isPresent) listOf(get()) else emptyList()
+    }
+    ```
+
+    <deflist collapsible="true">
+       <def title="Extension functions">
+          <p>The return type of <code>findById()</code> function in <code>CrudRepository</code> interface is an instance of <code>Optional</code> class. However, it would be convenient to return a <code>List</code> with a single message for consistency. For that, you need to unwrap the <code>Optional</code> value if it’s present, and return a list with the value. This can be implemented as an <a href="extensions.md#extension-functions">extension function</a> to the <code>Optional</code> type.</p>
```

Comment:
```suggestion
          <p>The return type of the <code>findById()</code> function in <code>CrudRepository</code> interface is an instance of <code>Optional</code> class. However, it would be convenient to return a <code>List</code> with a single message for consistency. For that, you need to unwrap the <code>Optional</code> value if it’s present, and return a list with the value. This can be implemented as an <a href="extensions.md#extension-functions">extension function</a> to the <code>Optional</code> type.</p>
```